networking_cisco/plugins/cisco/cfg_agent/device_drivers/asr1k/aci_asr1k_routing_driver.py

```diff
@@ -542,3 +538,38 @@ def _set_vrf_pid(self, ri, port, config):
         # TODO(tbachman): fix to ensure PID uniqueness
         pid = int(hashlib.md5(vrf).hexdigest()[:4], 16)
         return str(pid)
+
+    def _get_snat_pool_name(self, subnet):
+        is_multi_region_enabled = cfg.CONF.multi_region.enable_multi_region
+        if is_multi_region_enabled:
+            region_id = cfg.CONF.multi_region.region_id
+            snat_id = "%s-%s_nat_pool" % (subnet['id'][:self.DEV_NAME_LEN - 6],
```

Comment:
The original idea was just to keep the NAT pool naming convention consistent with the stock ASR implementation, but I think my implementation was inconsistent. The stock ASR implementation has a NAT pool per VRF, so VRF names are used for constructing the NAT pool name (A VRF name in the stock ASR implementation is 'nrouter-', followed by the first 6 characters of the router UUID [the DEV_NAME_LEN is 14, and 'nrouter-' is 8 characters], with an optional suffix of a "-" and a 6-character multi-region value [when multi-region is configured]). Since our mapping is configurable, we use the ID passed down in the SNAT dictionary. I guess my thinking was to add the 'nrouter-' prefix, but it looks like that got forgotten. This isn't a problem, but I can see the concern over the use of a magic numbers. I'll add a 'snat-id-' prefix to replace the 'nrouter-' prefix (same length), and create a new constant with this magic number, describing the reason.